tests/functional/test_show.py

```diff
@@ -79,6 +79,31 @@ def test_find_package_not_found():
     assert len(list(result)) == 0
 
 
+def test_report_single_not_found(script):
+    """
+    Test that a single unmatched package is reported not found.
+    This should report an error as there no result to print but should
+    also give a warning on stderr. N.B. may need allow_stderr_warning=True
```

Comment:
Looking at this now, rather than phrasing it in terms of changes that may or may not happen in the future, I think it would be better to say, as a code comment before the line, something like, "There is no need to pass allow_stderr_warning=True since this is implied by expect_error." I can tell from your discussion comments that you want this change to happen in the future, but I think it would be better to keep it out of the code comments until it's been discussed and agreed on.

Reply:
Done